src/adaptors/http1/http1_client.c

```diff
@@ -1238,6 +1505,24 @@ uint64_t qdr_http1_client_core_link_deliver(qdr_http1_adaptor_t    *adaptor,
     _client_response_msg_t *rmsg = DEQ_TAIL(hreq->responses);
     assert(rmsg && rmsg->dlv == delivery);
 
+    // when aggregating responses, they are saved on the list until
+    // the request has been settled, then encoded in the configured
+    // aggregation format
+    if (hconn->cfg.aggregation != QD_AGGREGATION_NONE) {
+        qd_log(qdr_http1_adaptor->log, QD_LOG_DEBUG, "[C%"PRIu64"][L%"PRIu64"] Received response (%i responses received), settling", hconn->conn_id, link->identity, DEQ_SIZE(hreq->responses));
+        rmsg->dispo = PN_ACCEPTED;
+        qd_message_set_send_complete(msg);
+        qdr_link_flow(qdr_http1_adaptor->core, link, 1, false);
+        qdr_delivery_remote_state_updated(qdr_http1_adaptor->core,
+                                          rmsg->dlv,
+                                          rmsg->dispo,
+                                          true,   // settled,
+                                          0,      // error
+                                          0,      // dispo data
+                                          false);
+        return PN_ACCEPTED;
```

Comment:
Since we are holding the response message buffers, I believe it may be necessary to disable Q2 on the arriving response messages by calling qd_message_Q2_holdoff_disable(). and possibly "restart" the receiving thread via qd_link_restart_rx(qd_message_get_receiving_link(msg)).  

See https://github.com/apache/qpid-dispatch/blob/master/src/message.c#L2578  - something like that. 

This could be tested by having the multicast responses contain > 512K bodies.

Oh what a tangled web we weave...

Reply:
I did test with large responses (1M) and it seems to work so while you may be right that something is needed here, I've left that for now.

Reply:
Yes let's not block this patch for that - we'll add a test case for it in a follow up.
We'll need a multi-hop router test case since Q2 is a proton layer thing.  I'd expect Q2 to not be a factor in router standalone configurations.